gatorgrade/output/output_functions.py

```diff
@@ -1,7 +1,57 @@
-"""Add docstring."""
+"""
+This class is used for storing the main functions requested from the Github.
 
+Issue Tracker for the output team.
+For instance, functions dealing with percentage output, description output,
+and colorization of text.
+"""
+import colorama as color
 
-# This class is used for storing the main functions requested
-# from the Github Issue Tracker for the output team.
-# For instance, functions dealing with percentage output,
-# description output, and colorization of text.
+color.init()
+
+
+def sort_checks_by_result(results):
+    """
+    Process results and determine if the check passed or failed.
+
+    Argument `results` is: list[(check_result_element1, check_result_element2),(...)]]
+    """
+    passed_checks = []
+    failed_checks = []
+    # iterate through results tuples
+    for i in results:
+        for j in i:
+            if isinstance(j, bool):
+                if j is True:
+                    passed_checks.append(i)
+                else:
+                    failed_checks.append(i)
+    output_passed_checks(passed_checks)
+    output_failed_checks(failed_checks)
+
+
+def output_passed_checks(passed_checks):
+    """Output the results for all of the checks that passed using the passed_checks list."""
+    for i in passed_checks:
+        requirement = i[0]
+        # Use colorama to style passing check
+        print(f"{color.Fore.GREEN}\u2714  {color.Style.RESET_ALL}{requirement}")
+
+
+def output_failed_checks(failed_checks):
+    """Output the results for all of the checks that did not pass using the failed_checks list."""
+    for i in failed_checks:
+        # Extract the details of each check
+        requirement = i[0]
+        description = i[2]

```

Comment:
I like the goal of clarity! As @Michionlion pointed out above, it would likely add more complexity to change the data types, however I did updated the for loops throughout the program to be much more descriptive. Meaning that in the code you pointed out above it now reads, `requirement = check[0]` and `description = check[2]`. I could also add comments to clarify exactly what information is being stored in those variables if necessary. What do you think?

Reply:
I think this change to define variables is GREAT!!! And then you don't even need comments to describe what's stored in `check[0]`, etc. The variable name does the work for you! Great change!